### PR TITLE
Upgrade to ioredis@5

### PR DESCRIPTION
**Pull request by jleverenz**
Gave the upgrade a shot, happy to make adjustments and help get this ready for merge / release.

### CODE REVIEW

File: lib/redis.interfaces.ts [hunk 1]
```diff
@@ -1,10 +1,8 @@
 import { ModuleMetadata, Type } from '@nestjs/common/interfaces';
-import * as Redis from 'ioredis';
-
-export type Redis = Redis.Redis;
+import { RedisOptions } from 'ioredis';
 
 export interface RedisModuleOptions {
-  config: Redis.RedisOptions & { url?: string };
```
Comment (zorndorff) [inline lib/redis.interfaces.ts:7]: There's a chance they had a specific reason to do it this way. `* as Redis` can fix compatibility issues with common module default exports.
Hopefully fixed with the ioredis upgrade.

Comment (zorndorff) [reply lib/redis.interfaces.ts:7]: Nice ! 
I guess the only other thing to test would be use with a plain js nest project? 
I think that's a possibility.

**Comment by gingermusketeer** [comment]
I would love to make use of this. Anything I can do to help get this change in?

**Comment by pradm007** [comment]
Hello All,
Any update on getting this change merged in? We wanted to use this change (`ioredis:^5.0.0`) for our project. Currently, there are conflicts with `ioredis:^4.0.0` and `ioredis@5.3.2` when used explicitly.

**Comment by aaronovz1** [comment]
Bump - also interested in this

**Pull request merged**